src/util/messages.ml

```diff
@@ -3,6 +3,173 @@ open Pretty
 open GobConfig
 module GU = Goblintutil
 
+type array_oob =
+  | PastEnd
+  | BeforeStart
+  | Unknown
+
+type undefined_behavior =
+  | ArrayOutOfBounds of array_oob
+  | NullPointerDereference
+  | UseAfterFree
+
+type behavior =
+  | Undefined of undefined_behavior
+  | Implementation
+  | Machine
+
+type integer = Overflow | DivByZero
+
+type cast = TypeMismatch
+
+type warning =
+  | Behavior of behavior
+  | Integer of integer
+  | Race
+  | Cast of cast
+  | Unknown of string
+  | Debug of string
+  | Analyzer
```

Comment:
What is the difference in meaning for `Debug` and `Analyzer`?

Reply:
I expect `Analyzer` to have subtypes later as I think that most current warnings are probably Analyzer warnings? Or at least I feel like most doesn't fall into any other categories.